calicoctl/calico_ctl/container.py

```diff
@@ -506,7 +506,8 @@ def get_ip_and_pool(ip):
             ip = result[1][0]
         pool = get_pool_or_exit(ip)
     elif ip is not None and '/' in ip:
-        pool = IPPool(ip)
+        cidr = IPNetwork(ip)
+        pool = get_pool_or_exit(IPAddress(cidr.first))
```

Comment:
Sorry I didn't spot this before.  Checking for just the first IP address is not sufficient here.  The API mentions that the CIDR must _exactly_ match the CIDR of a pool - so I think it is better to write another function to query the pools and compare the CIDR with that of the pool.  We so something similar in our libnetwork plugin (see below).  Could either update get_pool_or_exit() to accept a CIDR and do an exact match, or implement a separate method just for handling getting a pool from a CIDR (and exiting if it doesn't exist).

```
def get_pool(pool_cidr):
    """
    Return the Calico IP Pool with the specified pool CIDR, or None if not
    present.
    :param pool_cidr:  The pool CIDR.
    :return:  IPPool - The Calico pool, or None if not configured.
    """
    pools = client.get_ip_pools(pool_cidr.version, ipam=True)
    for pool in pools:
        if pool.cidr == pool_cidr:
            return pool
    return None
```

See https://github.com/projectcalico/libnetwork-plugin/blob/master/libnetwork/driver_plugin.py